src/script/miniscript.h

```diff
@@ -523,6 +523,20 @@ struct Node {
         }
     }
 
+    Node<Key> Clone() const
+    {
+        // Use TreeEval() to avoid a stack-overflow due to recursion
+        auto upfn = [](const Node& node, Span<Node> subs) {
+            Node<Key> ret(node);
+            ret.subs.clear();
+            for (const auto& sub : subs) {
+                ret.subs.push_back(MakeNodeRef<Key>(sub));
+            }
+            return ret;
+        };
+        return TreeEval<Node<Key>>(upfn);
+    }
```

Comment:
More smartpointy less segfaulty:
```diff
diff --git a/src/script/descriptor.cpp b/src/script/descriptor.cpp
index 6053c853bb..b099fdfc3f 100644
--- a/src/script/descriptor.cpp
+++ b/src/script/descriptor.cpp
@@ -1360,7 +1360,7 @@ public:
         for (const auto& arg : m_pubkey_args) {
             providers.push_back(arg->Clone());
         }
-        return std::make_unique<MiniscriptDescriptor>(std::move(providers), miniscript::MakeNodeRef<uint32_t>(m_node->Clone()));
+        return std::make_unique<MiniscriptDescriptor>(std::move(providers), m_node->Clone());
     }
 };
 
diff --git a/src/script/miniscript.h b/src/script/miniscript.h
index a74e312e79..1472c9ac4f 100644
--- a/src/script/miniscript.h
+++ b/src/script/miniscript.h
@@ -523,18 +523,18 @@ struct Node {
         }
     }
 
-    Node<Key> Clone() const
+    NodeRef<Key> Clone() const
     {
         // Use TreeEval() to avoid a stack-overflow due to recursion
-        auto upfn = [](const Node& node, Span<Node> subs) {
-            Node<Key> ret(node);
-            ret.subs.clear();
+        auto upfn = [](const Node& node, Span<NodeRef<Key>> subs) {
+            auto ret(MakeNodeRef<Key>(node));
+            ret->subs.clear();
             for (const auto& sub : subs) {
-                ret.subs.push_back(MakeNodeRef<Key>(sub));
+                ret->subs.push_back(sub);
             }
             return ret;
         };
-        return TreeEval<Node<Key>>(upfn);
+        return TreeEval<NodeRef<Key>>(upfn);
     }
 
 private:
```

Reply:
Thanks, taken.